wizard/src/components/mapping_section.tsx

```diff
@@ -0,0 +1,366 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * Component for showing and updating the mapping.
+ */
+
+import _ from "lodash";
+import React, { useEffect, useRef, useState } from "react";
+import { Button } from "reactstrap";
+
+import {
+  Column,
+  CsvData,
+  DCProperty,
+  DCType,
+  MappedThing,
+  Mapping,
+  MappingType,
+  MappingVal,
+} from "../types";
+import { PlaceDetector } from "../utils/detect_place";
+import { checkMappings } from "../utils/validation";
+import { MappingColumnOptions } from "./mapping_column_options";
+import { MappingConstantsSection } from "./mapping_constants_section";
+import { MappingSectionTable } from "./mapping_section_table";
+
+export interface ColumnInfo {
+  // the column of interest
+  column: Column;
+  // the mapped thing selected for column mapping type
+  columnMappedThing: MappedThing;
```

Comment:
why columnMappedThing and headerMappedThing are different?

Reply:
The selected input for columnMappedThing and headerMappedThing could be different: https://screenshot.googleplex.com/3AgXskBkmBXmxsJ